lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2341,8 +2341,36 @@ static bool computeIsSetterMutating(TypeChecker &TC,
   llvm_unreachable("bad storage kind");
 }
 
+static bool shouldUseOpaqueReadAccessor(TypeChecker &TC,
+                                        AbstractStorageDecl *storage) {
+  // If the attribute isn't present, do nothing.
+  if (!storage->getAttrs().hasAttribute<BorrowedAttr>())
+    return false;
+
+  // The start of this logic follows
+  // AbstractStorageDecl::requiresOpaqueModifyCoroutine().
+
+  // These criteria are basically required for correctness.
```

Comment:
Should they not be enforced via diagnostics instead then?

Reply:
Look at you having high expectations. :)